### ISSUE TITLE
[Question] WaveRNN Vocoder

### ISSUE THREAD

**Issue by sooftware** [Tomiinek/Multilingual_Text_to_Speech#37]
Hi! Tomiinek!!  
I trained tacotron with `generated-training.json` config.  
With your WaveRNN weight file, result of tts are pretty good.  
  
But WaveRNN that you opened was a pre-trained model in only 5 languages, so there were some disappointing results for other languages.  
So, I trained WaveRNN with your [wavernn repository](https://github.com/Tomiinek/WaveRNN).  
With your comment in [here](https://github.com/Tomiinek/Multilingual_Text_to_Speech/issues/31), I trained WaveRNN.  
  
I've trained WaveRNN with 600k steps, but this WaveRNN doesn't make voice at all.  
So I confused. The WaveRNN you opened is good at producing voice, but why can't I do it at all?  
  
<img src="https://user-images.githubusercontent.com/42150335/102718656-9b6a7780-432c-11eb-8116-e694be502a78.png" width=500>
  
The figure above is a waveform made from my WaveRNN.
  
<img src="https://user-images.githubusercontent.com/42150335/102719754-35cdb980-4333-11eb-8b52-19676edbffc7.png" width=500>
  
The figure above is a waveform made from your WaveRNN.
  
The training loss scale is about 2.3. Here are sample quant and mel spectrogram. [Quant Sample](https://drive.google.com/file/d/195ZluLjB4DRcheG1es1JzAihxuCs2RWA/view?usp=sharing) [Mel Sample](https://drive.google.com/file/d/1AiSfV2Ag6ZGeShaZZkgrCXxlCdct8Ec4/view?usp=sharing).   
I think if there is a Mel-Spectrogram, Wave pair, it should be trained, but I don't know why not.  
Below figures are image of Quant Sample and Mel Sample.  
  
<img src="https://user-images.githubusercontent.com/42150335/102720347-26e90600-4337-11eb-92c9-84d39de0de72.png" width=600>
  
<img src="https://user-images.githubusercontent.com/42150335/102720356-323c3180-4337-11eb-807a-3970827685c5.png" width=600>

  
I did not touch the parameters of hparams.py in the WaveRNN repository. Could you check it for me?   
Please help me Tomiinek.
Thank you.

**Comment by Tomiinek** [comment]
Hello, I am sorry for a late response. 

1) I have problems with training the model too and it was very very slow. You can always try out different vocoders like MelGAN or WaveGlow (I do not know what is SOTA currently).

2) I have two questions. Do the training audios have the same scale and are they normalized to the same volume? Are the quant values up to 1000 ok?

**Comment by sooftware** [comment]
I'll check the scale of audio. Is the frame length of your tachotron melspectrogram 50 ms and frame shift 12.5 ms?  
Also, Is there a WaveRNN that I learned with Lj speech?  
  
Thank you for your response.

**Comment by sooftware** [comment]
Audio's scale is the same. (up to 1000)

**Comment by sooftware** [comment]
Furthermore, if I insert the pre-built mel (from gta.py) into the WaveRNN you opened as input, it produces no speech at all.

**Comment by Tomiinek** [comment]
> Is the frame length of your tacotron Mel spectrogram 50 ms and frame shift 12.5 ms?

Yes, it is.

> Also, Is there a WaveRNN that I learned with Lj speech?

I think that it is in the original repository that I forked.

> Audio's scale is the same. (up to 1000)

Ok, and is the scale the same if you apply some volume normalization such as `sox infile.wav outfile.wav gain −n −3`? I would really expect audio scale in [-1, 1].

> Furthermore, if I insert the pre-built mel (from gta.py) into the WaveRNN you opened as input, it produces no speech at all.

I do not understand, can you please explain in more detail?

**Comment by sooftware** [comment]
Hmm... Thank you for your answer!!  
Never mind the last question. I'll think about it a little more.
